lib/spack/spack/cmd/stage.py

```diff
@@ -50,4 +50,6 @@ def stage(parser, args):
     specs = spack.cmd.parse_specs(args.specs, concretize=True)
     for spec in specs:
         package = spack.repo.get(spec)
-        package.do_stage()
+        with package.stage as stage:
+            stage.delete_on_exit = False
+            package.do_stage()
```

Comment:
The idea with the original `do_*` calls on `Package` was to keep the interface exposed to commands really simple.  I think this part, especially since you have to refer to the stage variable explicitly, is kind of a leaky abstraction.  I'd rather the `package.py` encapsulate the hard part... Is there a reason to require everyone who deals with `Package` to know the details of how staging is implemented?  Seems like too much to expect from the caller.